File: Try_GameEngine5/src/com/example/try_gameengine/action/MovementAtionController.java
package com.example.try_gameengine.action;


/**
 * MovementAtionController can control some work.
 * @author irons
 *
 */
public class MovementAtionController {
	MovementAction action;
	
	/**
	 * @param action
	 */
	public void setMovementAction(MovementAction action){
		this.action = action;
	}
	
	/**
	 * 
	 */
	public void cancelCurrentMove(){
		action.cancelMove();
	}
	
	/**
	 * 
	 */
	public void cancelAllMove(){
		action.cancelAllMove();
	}
	
	/**
	 * 
	 */
	public void pause(){
		action.pause();
	}
	
	public void pause(long milliseconds){
		action.pause();
	}
	
	/**
	 * 
	 */
	public void resume(){
		action.pause();
	}
	
	/**
	 * 
	 */
	public void restart(){
		action.cancelMove();
		action.start();
	}
	
	/**
	 * @param loop
	 */
	public void looper(boolean loop){
		if(action.isFinish()){
			
		}
	}
}

File: Try_GameEngine5/src/com/example/try_gameengine/action/MovementAtionController.kt
package com.example.try_gameengine.action


/**
 * MovementAtionController can control some work.
 * @author irons
 // */
class MovementAtionController {
    var action: MovementAction? = null

    /**
     * @param action
     // */
    fun setMovementAction(action: MovementAction) {
        this.action = action
    }

    /**
     * 
     // */
    fun cancelCurrentMove() {
        action!!.cancelMove()
    }

    /**
     * 
     // */
    fun cancelAllMove() {
        action!!.cancelAllMove()
    }

    /**
     * 
     // */
    fun pause() {
        action!!.pause()
    }

    fun pause(milliseconds: Long) {
        action!!.pause()
    }

    /**
     * 
     // */
    fun resume() {
        action!!.pause()
    }

    /**
     * 
     // */
    fun restart() {
        action!!.cancelMove()
        action!!.start()
    }

    /**
     * @param loop
     // */
    fun looper(loop: Boolean) {
        if (action!!.isFinish()) {
        }
    }
}
